korge/src/commonMain/kotlin/com/soywiz/korge/view/Sprite.kt

```diff
@@ -118,28 +147,38 @@ open class Sprite(
 
     private fun nextSprite(frameTime : TimeSpan){
         lastAnimationFrameTime+=frameTime
-        if ((animationCyclesRequested > 0 || animationRequestedDuration > 0.milliseconds || animationLooped) && lastAnimationFrameTime+frameTime >= this.spriteDisplayTime){
-            bitmap = currentAnimation?.getSprite(++currentSpriteIndex) ?: Bitmaps.transparent
-            animationCyclesRequested--
+        if ((animationNumberOfFramesRequested > 0 || animationRequestedDuration > 0.milliseconds || animationLooped) && lastAnimationFrameTime+frameTime >= this.spriteDisplayTime){
+            showBitmapAtIndex(if (reversed) --currentSpriteIndex else ++currentSpriteIndex)
+            animationNumberOfFramesRequested--
             animationRequestedDuration-=(frameTime+spriteDisplayTime)
             lastAnimationFrameTime = 0.milliseconds
         }
     }
 
     private fun updateCurrentAnimation(
-        spriteAnimation: SpriteAnimation,
+        spriteAnimation: SpriteAnimation?,
         spriteDisplayTime: TimeSpan = this.spriteDisplayTime,
         animationCyclesRequested : Int = 1,
         duration : TimeSpan = 0.milliseconds,
-        looped : Boolean = false
+        startFrameIndex: Int = 0,
+        looped : Boolean = false,
+        reversed : Boolean = false
     ){
         triggerEvent(onAnimationStarted)
         this.spriteDisplayTime = spriteDisplayTime
         currentAnimation = spriteAnimation
         animationRequested = true
         animationLooped = looped
         animationRequestedDuration = duration
-        this.animationCyclesRequested = if (!looped) animationCyclesRequested else 1
+        currentSpriteIndex = startFrameIndex
+        this.reversed = reversed
+        currentAnimation?.let {
+            this.animationNumberOfFramesRequested = if (!looped) it.spriteStackSize-1 else 1
+        }
+    }
+
+    fun showBitmapAtIndex(index : Int)  {
```

Comment:
What do you think about `setFrame` or `setAnimationFrame`? The shortest thing we can find that still understands, the better

Reply:
`showFrameAt(index : Int)` ?

Reply:
Or just as property
`frame = 5`

Reply:
That's another option. As you prefer